.github/workflows/rust.yml

```diff
@@ -96,6 +96,19 @@ jobs:
         command: test
         args: --all-features
 
+  test-miri:
+    name: Test Miri
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install Miri
+      run: |
+        rustup toolchain install nightly --component miri
```

Comment:
for some reason the miri component doesn't get properly installed when using `action-rs/toolchain` so I just do it manually